app/main.py

```diff
@@ -1,3 +1,59 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(
+            self, other: Distance | int | float
+    ) -> Distance:
+        if isinstance(other, int | float):
+            self.km = self.km + other
+        else:
+            self.km = self.km + other.km
+        return self
```

Comment:
`__add__` should return a new Distance instance, it should not override km attribute